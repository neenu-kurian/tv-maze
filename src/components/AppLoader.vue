<template>
  <div class="flex flex-col items-center justify-center mt-24">
    <div
      class="w-16 h-16 border-4 border-gray-400 rounded-full animate-spinner bg-white mx-auto"
      data-testid="app-loader"
    ></div>
    <p class="text-center mt-4" v-if="message" data-testid="loading-message">{{ message }}</p>
  </div>
</template>

<script setup lang="ts">
import type { AppLoaderProps } from '@/types/index.ts';

defineProps<AppLoaderProps>();
</script>

<style scoped>
@keyframes spinner {
  to {
    transform: rotate(360deg);
  }
}
</style>
