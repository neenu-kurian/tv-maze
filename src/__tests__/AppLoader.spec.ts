import { render } from '@testing-library/vue';
import AppLoader from '@/components/AppLoader.vue';
import { describe, it, expect } from 'vitest';

describe('AppLoader.vue', () => {
  it('matches snapshot', () => {
    const { container } = render(AppLoader, {
      props: {
        message: 'Loading...',
      },
    });
    expect(container).toMatchSnapshot();
  });
});
