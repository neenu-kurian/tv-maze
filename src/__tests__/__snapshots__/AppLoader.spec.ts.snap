// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`\`.vue > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center mt-24"
    data-v-8ec46d9a=""
  >
    <div
      class="w-16 h-16 border-4 border-gray-400 rounded-full animate-spinner bg-white mx-auto"
      data-testid="app-loader"
      data-v-8ec46d9a=""
    />
    <p
      class="text-center mt-4"
      data-testid="loading-message"
      data-v-8ec46d9a=""
    >
      Loading...
    </p>
  </div>
</div>
`;

exports[`AppLoader.vue > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center mt-24"
    data-v-8ec46d9a=""
  >
    <div
      class="w-16 h-16 border-4 border-gray-400 rounded-full animate-spinner bg-white mx-auto"
      data-testid="app-loader"
      data-v-8ec46d9a=""
    />
    <p
      class="text-center mt-4"
      data-testid="loading-message"
      data-v-8ec46d9a=""
    >
      Loading...
    </p>
  </div>
</div>
`;

exports[`HomeView.vue > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center mt-24"
    data-v-8ec46d9a=""
  >
    <div
      class="w-16 h-16 border-4 border-gray-400 rounded-full animate-spinner bg-white mx-auto"
      data-testid="app-loader"
      data-v-8ec46d9a=""
    />
    <p
      class="text-center mt-4"
      data-testid="loading-message"
      data-v-8ec46d9a=""
    >
      Loading...
    </p>
  </div>
</div>
`;
